### PR TITLE
main - update cli to fix ssh2 vulnerability

### PR DESCRIPTION
**Pull request by JillieBeanSim**
## Proposed changes

update zowe cli dependency to 7.21.1

## Release Notes

Milestone: 2.14.0

Changelog: Update dependencies to address vulnerabilities.

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [ ] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [ ] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

File: packages/zowe-explorer/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ All notable changes to the "vscode-extension-for-zowe" extension will be documen
 
 - Fixed the allocate-like functionality by removing the inclusion of DS item in the filter history. [#2620](https://github.com/zowe/vscode-extension-for-zowe/issues/2620)
 - Fixed issue with `Submit JCL` losing focus on JCL being submitted, causing the wrong job submission. [#2616](https://github.com/zowe/vscode-extension-for-zowe/issues/2616)
+- Update dependencies to address vulnerabilities.
```
Comment (zFernand0) [inline packages/zowe-explorer/CHANGELOG.md:13]: Do we want to call out the specific version where vulnerabilities were fixed, or should we just say something along the lines of "Updating dependencies for technical currency"?

**Review by zFernand0 — COMMENTED** [review]
LGTM! 😋 

Curious if we should do the same for our V1 version 😋

File: packages/zowe-explorer/CHANGELOG.md (hunk 1 above)
Comment (JillieBeanSim) [reply packages/zowe-explorer/CHANGELOG.md:13]: working on PRs for other branches with the versions Timothy shared

**Review by zFernand0 — CHANGES_REQUESTED** [review]
I don't think we should call out that this version fixes vulnerabilities 😢 

Could we make the statement a bit more vague and just say something about technical currency?

**Comment by JillieBeanSim** [comment]
> I don't think we should call out that this version fixes vulnerabilities 😢
> 
> Could we make the statement a bit more vague and just say something about technical currency?

I'm not sure stating that we fixed dependencies with vulnerabilities is a bad thing since it makes users aware that they need to update asap, but changed changelogs as requested.

**Review by adam-wolfe — DISMISSED** [review]
LGTM

**Comment by zFernand0** [comment]
> > I don't think we should call out that this version fixes vulnerabilities 😢
> > Could we make the statement a bit more vague and just say something about technical currency?
> 
> I'm not sure stating that we fixed dependencies with vulnerabilities is a bad thing since it makes users aware that they need to update asap, but changed changelogs as requested.

It also lets people know what to look for when trying to cause some damage 😅

**Review by zFernand0 — APPROVED** [review]

**Comment by adam-wolfe** [comment]
Just for reference: https://www.zowe.org/security.html#security-issues-disclosure

**Comment by JillieBeanSim** [comment]
> Just for reference: https://www.zowe.org/security.html#security-issues-disclosure

FYI @anaxceron ☝🏻

**Review by zFernand0 — APPROVED** [review]

**Review by adam-wolfe — APPROVED** [review]

**Review by t1m0thyj — APPROVED** [review]

**Pull request merged**